server/storage/rethinkdb.go

```diff
@@ -231,6 +232,28 @@ func (rdb RethinkDB) GetChecksum(gun, role, checksum string) (created *time.Time
 	return &file.CreatedAt, file.Data, err
 }
 
+// GetVersion gets a specific TUF record by its version
+func (rdb RethinkDB) GetVersion(gun, role, version string) (*time.Time, []byte, error) {
+	var file RDBTUFFile
+	v, err := strconv.Atoi(version)
+	if err != nil {
+		return nil, nil, ErrNotFound{}
+	}
+	res, err := gorethink.DB(rdb.dbName).Table(file.TableName(), gorethink.TableOpts{ReadMode: "majority"}).Get([]interface{}{gun, role, v}).Run(rdb.sess)
+	if err != nil {
+		return nil, nil, err
+	}
+	defer res.Close()
+	if res.IsNil() {
+		return nil, nil, ErrNotFound{}
+	}
+	err = res.One(&file)
+	if err == gorethink.ErrEmptyResult {
```

Comment:
Non-blocking:  just out of curiosity, can this happen, since you checked for `nil` before?